src/engine/strat_engine/backstore/crypt/handle.rs

```diff
@@ -309,14 +322,20 @@ impl CryptHandle {
         dev_uuid: DevUuid,
         pool_name: &Name,
         encryption_info: &EncryptionInfo,
+        luks2_params: Option<&CryptParamsLuks2>,
     ) -> StratisResult<()> {
+        let mut luks2_params_ref: Option<CryptParamsLuks2Ref<'_>> = luks2_params.map(|lp| {
+            lp.try_into()
+                .expect("the luks2_param struct was constructed by stratisd, so it is valid")
```

Comment:
I would prefer to return an error here just in case there's ever breakage somewhere in the stack. We already have the capability in this function to return an error, and I think an initialization error would be better than taking stratisd down.

Reply:
Ok!